SpokestackTests/SpeechPipelineTest.swift

```diff
@@ -30,142 +31,195 @@ class SpeechPipelineTest: XCTestCase {
-        wait(for: [didInitExpectation], timeout: 1)
-        XCTAssert(p.status())
-        delegate = SpeechPipelineTestDelegate()
-        delegate.asyncExpectation = didInitExpectation
-        XCTAssert(p.status())
-    }
-    
-    /// setDelegates
-    func testSetDelegates() {
-        var delegate: SpeechPipelineTestDelegate
-        let didInitExpectation = expectation(description: "didInitExpectation fulfills when testSetDelegates calls SpeechPipelineTestDelegate as the result of didInit method completion")
-        let activateExpectation = expectation(description: "activateExpectation fulfills when testSetDelegates calls SpeechPipelineTestDelegate as the result of activate method completion")
-        
-        // init the pipeline
-        delegate = SpeechPipelineTestDelegate()
-        delegate.asyncExpectation = didInitExpectation
-        let p = SpeechPipeline(TestProcessor(true), speechConfiguration: SpeechConfiguration(), speechDelegate: delegate, wakewordService: TestProcessor(), pipelineDelegate: delegate)
-        wait(for: [didInitExpectation], timeout: 1)
-        
-        // change the pipeline's delegates
-        delegate = SpeechPipelineTestDelegate()
-        delegate.asyncExpectation = activateExpectation
-        p.setDelegates(delegate, pipelineDelegate: delegate)
-        
-        // assert that the pipeline's delegate references (and pipeline's service delegates) have changed
-        XCTAssert(delegate === p.speechDelegate)
-        p.speechDelegate?.didActivate()
-        wait(for: [activateExpectation], timeout: 1)
-        XCTAssert(delegate.activated)
+        XCTAssert(p.configuration.fftHopLength == 30)
     }
     
     /// activate & deactivate
     func testActivateDeactivate() {
         let didInitExpectation = expectation(description: "didInitExpectation fulfills when testActivateDeactivate calls SpeechPipelineTestDelegate as the result of didInit method completion")
+        let didActivateExpectation = expectation(description: "didActivateExpectation fulfills when testActivateDeactivate calls SpeechPipelineTestDelegate as the result of activate method completion")
+        let didDeActivateExpectation = expectation(description: "didDeActivateExpectation fulfills when testActivateDeactivate calls SpeechPipelineTestDelegate as the result of deactive method completion")
         let delegate = SpeechPipelineTestDelegate()
-        delegate.asyncExpectation = didInitExpectation
         let config = SpeechConfiguration()
+        let context = SpeechContext()
 
         // init the pipeline
-        let p = SpeechPipeline(TestProcessor(true), speechConfiguration: config, speechDelegate: delegate, wakewordService: TestProcessor(), pipelineDelegate: delegate)
+        delegate.asyncExpectation = didInitExpectation
+        let p = SpeechPipeline(configuration: config, listeners: [delegate])
+        _ = TestProcessor(true, config: config, context: context)
```

Comment:
why is this initialized and thrown away?

Reply:
i…don't remember. thanks for pointing that out.